presto-main-base/src/main/java/com/facebook/presto/execution/StageExecutionStats.java

```diff
@@ -203,8 +203,10 @@ public StageExecutionStats(
 
         checkArgument(bufferedDataSizeInBytes >= 0, "bufferedDataSizeInBytes is negative");
         this.bufferedDataSizeInBytes = bufferedDataSizeInBytes;
-        checkArgument(outputDataSizeInBytes >= 0, "outputDataSizeInBytes is negative");
-        this.outputDataSizeInBytes = outputDataSizeInBytes;
+
+        // An overflow could have occurred on this stat - handle this gracefully.
+        this.outputDataSizeInBytes = (outputDataSizeInBytes >= 0) ? outputDataSizeInBytes : Long.MAX_VALUE;
```

Comment:
should we do this for all the argument checks?

Reply:
@rschlussel 
Let's do it for only this 1st. Bytes are much bigger magnitude than, say, rows.
Later we can extend it for other bytes-related metrics.